### DISCUSSION TITLE
Can you have wire:model.live and wire:model.blur at the same time?

### DISCUSSION THREAD

**Discussion by msurdi** [livewire/livewire#8050] [1. Help] [answered]
Hi!

I'm new to Livewire and just trying it out for a side project, sorry on beforehand if I'm asking something obvious.

I have a use case where I want to use `wire.model.live.debounce=3000ms` (quite a long time but I only do want to validate this input after the user has definitely stopped typing given it's a pretty expensive operation) but at the same time I also want the validation to run right away if the user tabs away from the input.

I've found that adding both attributes to the element, that is ` wire:model.blur="my_field"` and at the same time `wire:model.live.debounce.3000ms="my_field"` then things works as I expect, but 1) It doesn't seem that clean as I have to duplicate the field name in the attribute's value and 2) I'm not sure if this could cause any issues I'm not aware of.

Of course I've tried several combinations of ` wire:model.blur.live.debounce.3000s="my_value"` and similar combinations but just one of them seem to work each time, depending on the order.

Any ssuggestion?

Thanks!

**Comment by danie-ramdhani** [comment]
why not use `wire:model.live.lazy` (works when you blur away from input and when user press enter)?

**Reply by msurdi** [reply]
Thanks for the suggestion, but that doesn't work (the `live` part of it  doesn't run).

**Comment by msurdi** [comment, marked as answer]
Answering myself for future me (and others):

While the approach I was following of adding two `wire:model...` props worked, it had the drawback that there was always an extra (unnecessary) network call: If the component refresh happened due to a blur right after the value was changed, after the debounce time a second network call would also trigger due to the "live" watching, despite the value was the exact same at "blur" time.

I've finally solved this and got the exact desired behavior I wanted by creating the following Alpine component:

```JavaScript
Alpine.data("input", () => ({
    previousValue: "",
    init() {
        this.previousValue = this.$el.value;
    },
    refreshIfChanged(event) {
        const { value } = event.target;
        if (value !== this.previousValue) {
            this.previousValue = value;
            this.$wire.$refresh();
        }
    },
}));
```

And finally using it in my Livewire component like:

```html
<input x-data="input" x-on:blur="refreshIfChanged" x-on:input.debounce.1000ms="refreshIfChanged" wire:model="my_field" ... >
```
